### PR TITLE
systemd-homed: various fixups

### PR DESCRIPTION
**Pull request by WavyEbuilder**
Commit messages for each commit should explain more, let me know if you have any more questions.

Thanks!

### CODE REVIEW

File: policy/modules/system/systemd.if [hunk 1]
```diff
@@ -1136,6 +1136,25 @@ interface(`systemd_dbus_chat_homed',`
...
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`systemd_stream_connect_homed',`
+	gen_require(`
+		type systemd_homed_t;
+	')
+
+	allow $1 systemd_homed_t:unix_stream_socket connectto;
```
Comment (pebenito) [inline policy/modules/system/systemd.if:1155]: The interface summary says this is a named socket, so there should be `dir` and `sock_file` access too, preferably using `stream_connect_pattern()`.

File: policy/modules/system/systemd.fc [hunk 2]
```diff
@@ -61,6 +61,11 @@ HOME_DIR/\.config/containers/systemd(/.*)?		gen_context(system_u:object_r:system
 HOME_DIR/\.config/systemd(/.*)?		gen_context(system_u:object_r:systemd_conf_home_t,s0)
 HOME_DIR/\.local/share/systemd(/.*)?		gen_context(system_u:object_r:systemd_data_home_t,s0)
 
+# homed files
+HOME_DIR/\.identity	--	gen_context(system_u:object_r:systemd_homed_record_t,s0)
+HOME_DIR/\.identity-blob(/.*)?	gen_context(system_u:object_r:systemd_homed_record_t,s0)
```
Comment (pebenito) [inline policy/modules/system/systemd.fc:66]: Are these intended/required to be inaccessible to the user?

File: policy/modules/system/systemd.te [hunk 3]
```diff
@@ -725,14 +735,21 @@ allow systemd_homework_t systemd_homed_runtime_t:dir manage_dir_perms;
...
+allow systemd_homework_t systemd_homed_runtime_work_dir_t:dir mounton;
 
 allow systemd_homework_t systemd_homed_storage_t:file manage_file_perms;
 files_home_filetrans(systemd_homework_t, systemd_homed_storage_t, file)
 
 allow systemd_homework_t systemd_homed_tmpfs_t:file rw_inherited_file_perms;
 
+# setup luks backed home directories in /run/cryptsetup
+lvm_manage_runtime_files(systemd_homework_t)
+lvm_manage_runtime_dirs(systemd_homework_t)
```
Comment (pebenito) [inline policy/modules/system/systemd.te:751]: This should be made optional.

File: policy/modules/system/systemd.te [hunk 4]
```diff
@@ -769,6 +786,10 @@ systemd_log_parse_environment(systemd_homework_t)
 
 udev_read_runtime_files(systemd_homework_t)
 
+read_files_pattern(systemd_homework_t, systemd_homed_runtime_work_dir_t, systemd_homed_record_t)
+create_files_pattern(systemd_homework_t, systemd_homed_runtime_work_dir_t, systemd_homed_record_t)
+delete_files_pattern(systemd_homework_t, systemd_homed_runtime_work_dir_t, systemd_homed_record_t)
```
Comment (pebenito) [inline policy/modules/system/systemd.te:791]: This block should be up just above the machine-id hunk you added.

**Review by pebenito — CHANGES_REQUESTED** [review]
Also some lint issues in the github actions.

File: policy/modules/system/systemd.te (hunk 4 above)
Comment (WavyEbuilder) [reply policy/modules/system/systemd.te:791]: Got it, will do

File: policy/modules/system/systemd.te (hunk 3 above)
Comment (WavyEbuilder) [reply policy/modules/system/systemd.te:751]: Will do

File: policy/modules/system/systemd.if (hunk 1 above)
Comment (WavyEbuilder) [reply policy/modules/system/systemd.if:1155]: Will do

File: policy/modules/system/systemd.fc (hunk 2 above)
Comment (WavyEbuilder) [reply policy/modules/system/systemd.fc:66]: I did actually consider that. Looking at accountsservice and the various other things that are working with homed, I think the preferred way of getting account info is from the system userdb stuff. It has sensitive data (like the hashed password for that user) which is why I'm hesitant to make it readable by a user domain, but it is just for their own user. However, I can't see any reason for that information to be read or written to by the user, infact the records are signed so a user cannot write to it anyway. I'm happy to make it readable by the user if you wish however

File: policy/modules/system/systemd.te (hunk 3 above)
Comment (WavyEbuilder) [reply policy/modules/system/systemd.te]: Done

File: policy/modules/system/systemd.if (hunk 1 above)
Comment (WavyEbuilder) [reply policy/modules/system/systemd.if]: Done

File: policy/modules/system/systemd.if [hunk 5]
```diff
@@ -1152,7 +1152,7 @@ interface(`systemd_stream_connect_homed',`
 		type systemd_homed_t;
 	')
 
-	allow $1 systemd_homed_t:unix_stream_socket connectto;
+	stream_connect_pattern($1, systemd_userdbd_runtime_t, systemd_userdbd_runtime_t, systemd_homed_t)
```
Comment (pebenito) [inline policy/modules/system/systemd.if:1155]: Not all the types are in the `gen_require`.

File: policy/modules/system/systemd.te [hunk 6]
```diff
@@ -747,8 +751,10 @@ files_home_filetrans(systemd_homework_t, systemd_homed_storage_t, file)
 allow systemd_homework_t systemd_homed_tmpfs_t:file rw_inherited_file_perms;
 
 # setup luks backed home directories in /run/cryptsetup
-lvm_manage_runtime_files(systemd_homework_t)
-lvm_manage_runtime_dirs(systemd_homework_t)
+optional_policy(`
+	lvm_manage_runtime_files(systemd_homework_t)
+	lvm_manage_runtime_dirs(systemd_homework_t)
+')
```
Comment (pebenito) [inline policy/modules/system/systemd.te:757]: This should also move to the bottom of the `systemd_homework_t` block.

**Review by pebenito — CHANGES_REQUESTED** [review]

File: policy/modules/system/systemd.fc (hunk 2 above)
Comment (pebenito) [reply policy/modules/system/systemd.fc:66]: No, that's fine. I'm simply making sure I understand the intent.

**Review by pebenito — APPROVED** [review]

**Pull request merged**